copyspecial.py

```diff
@@ -13,29 +13,65 @@
 import commands
 import argparse
 
+
 """Copy Special exercise
 """
 
-# +++your code here+++
-# Write functions and modify main() to call them
 
-def main():
-    # This snippet will help you get started with the argparse module.
+def get_absolute_file_paths(directory):
+    regex = r"(_{2,}[^_\W]+_{2,})"
+    result = [os.path.realpath(x) for x in os.listdir(
+        directory) if re.search(regex, x)]
+    return result
+
+
+def copy_files_to_dir(files, new_directory, current_directory):
+
+    current_real_path = os.path.realpath(
+        current_directory) + '/' + str(new_directory)
+
+    if new_directory not in os.listdir(current_directory):
```

Comment:
This works in the example case, but this function contains a bug. Try running the same command twice and you'll see the exception that the file exists. This is because your gaurd is checking the current directory if the new directory exists. os.listdir only returns the directory names in the current directory though. In this case it's checking if tmp/fooby exists in ['CODEOWNERS', 'README.md', 'copyspecial.py', '.solutions', '.gitignore', '.git', 'straw_hat_random_owl.jpg', 'zz__something__.jpg', 'tmp', 'xyz__hello__.txt']

Since tmp/fooby doesn't exist in the list above, os.makedir creates it, but subsequent attempts fail because the check doesn't see the new directory despite it existing.